tests/README.md

```diff
@@ -0,0 +1,285 @@
+# VS Code Extension Tests
+
+## Folder structure
+
+- `unit` directory contains the a few utility functions that are used by the unit tests co-located
+  with the production code in the [`src/`](../src) directory.
+- `stubs` directory contains the stubs for the production code that are used by the unit tests.
+- `e2e` as the name suggests, contains the end-to-end (E2E) tests.
+- `fixtures` directory contains the fixtures for all tests. Use the `loadFixture` function in
+  [`fixtures/utils.ts`](fixtures/utils.ts) to read a particular fixture file.
+
+## E2E Tests using Playwright
+
+The E2E tests use [Playwright with Electron](https://playwright.dev/docs/api/class-electron) to
+launch VS Code and interact with it programmatically.
+
+### E2E Folder structure
+
+- `playwright.config.ts` file contains the configuration for the e2e tests. (Note: This is not to be
+  confused with the `playwright.config.ts` file in the root of the project. The root configuration
+  file is used to run the `*.spec.ts` files in the `src` directory for webview-specific testing via
+  `gulp functional`.).)
+- `specs` directory contains the test files.
+- `specs/utils` directory contains utility functions for the tests.
+
+> [!NOTE] What's the difference between `gulp e2e` and `gulp functional`?
+>
+> `gulp e2e` runs the E2E tests in the `tests/e2e/specs` directory, whereas `gulp functional` runs
+> the functional tests under `src/webview/` (and its subdirectories). Both use Playwright to run the
+> tests. However, the E2E tests run in a OS-native Electron window rather than just testing web
+> views in a browser environment.
+
+### Running the E2E tests
+
+> [!IMPORTANT] Please close any already-open VS Code windows before running the tests. This is
+> because the tests launch their own VS Code instance and need to properly handle browser auth
+> callbacks.
+
+We currently run the tests against Confluent Cloud production environment. We may add support for
+running against other non-production environments in the future.
+
+#### Pre-requisites
+
+Tests that interact with a Confluent Cloud resource (e.g. submitting a Flink statement) require you
+to have a `.env` file with the following environment variables:
```

Comment:
I might have missed it but should we mention folks need to install/init playwright before being able to run the tests?

Reply:
IIRC, I ran `npm install` before I tried to execute the tests. I might have missed something else 🤷‍♂️

Reply:
Yep I just realized that `npx playwright install` needs to be run separately.